### PR TITLE
72 connecting sb to external db

### PR DESCRIPTION
**Pull request by jhinkey**
Hi Preston,

Would you check these steps on using a DataSourceProvider JDK SPI to connect Service Builder to an external data source?

### CODE REVIEW

File: developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    file in your `*-service` module. For example, this file registers the
+    `DataSourceProvider` implementation from the previous step.
+
+```
+com.liferay.external.data.source.test.internal.DataSourceProviderImpl
+````
+
+5.  [Run Service Builder](/docs/7-2/appdev/-/knowledge_base/a/running-service-builder). 
+
+6.  [Deploy](/docs/7-2/reference/-/knowledge_base/r/deploying-a-project)
```
Comment (Preston-Crary) [inline developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: The DataSourceProvider, must be in the same -service module as the services consuming it.  We don't have a way to target different bundles and it's only applied to the one module.

File: developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    your `-service` module. If your `DataSourceProvider` is in a 
+    different project, deploy it too. 
+
+Now your module's Service Builder services persist data to the external data
```
Comment (Preston-Crary) [inline developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: This comes with a hidden cost.  Transactions between separate DataSources cannot be fully nested.  So depending on how the module with an external DataSource depends on other portal services rollbacks may not propagate between them.

Comment (Preston-Crary) [reply developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: If nested transactions are a requirement than using an external DataSource may not be appropriate.

Comment (Preston-Crary) [reply developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: @jhinkey we should give them this generic warning.  I don't want to go down into the details of what business logic can / cannot work - it's something they'll need to decide for themselves.  We can only give them a little warning so they know it's something to think about with their design / requirements.

Comment (jhinkey) [reply developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: @Preston-Crary - How's a note like this ...

**Warning:** If your Service Builder services require nested transactions, 
using an external data source may not be appropriate for you. Transactions
between separate data sources cannot be fully nested. Rollbacks may not
propagate between a module that uses an external data source and @product@
services (or another app's services) that use a different data source.

Comment (Preston-Crary) [reply developer/appdev/articles/03-service-builder/03-creating-o-r-map/98-connecting-service-builder-to-an-external-database.markdown]: Perfect!

**Pull request closed without merging**